src/virtualenv/activation/via_template.py

```diff
@@ -34,7 +34,7 @@ def generate(self, creator):
 
     def replacements(self, creator, dest_folder):
         current_platform = sysconfig.get_platform()
-        platforms = ["mingw", "cygwin", "msys"]
+        platforms = ["mingw", "msys"]
```

Comment:
I can remove this section of code entirely if you prefer, just wondered if this would be safer.

Reply:
I mean someone needs to check if mings, msys works same/different, and if is there a cygpath there. There're also tests you need to ammend/remove.

Reply:
I download MinGW/MSYS -- as I understand it, MSYS is an environment built on top of the MinGW tools (http://www.mingw.org/wiki/MSYS). The string that gets returned from `sysconfig.get_platform()` depends on which flavour you install, but they are all different versions of the same thing.

Running it locally, it seems that MSYS wants paths in the form `/c/somePath`, and it does not understand Cygwin-style paths:

```
dn_co@DESKTOP-1TDKNNU /c/Users/dn_co
$ cd /cygdrive/c/
sh: cd: /cygdrive/c/: No such file or directory
```
So I think the current implementation is good for MSYS/MinGW.

Reply:
It would suffer though from the same problem as cygwin does. Does msys has it's own variant of cygpath that we should be using to offer same guarantees we're offering for cygwin?

Reply:
https://stackoverflow.com/a/12063651 has some detailed answer to this

Reply:
👍 I can confirm that MSYS has cygpath, so we can approach it in the same way.